src/components/FormField/Select/index.tsx

```diff
@@ -0,0 +1,216 @@
+import React, { HTMLAttributes, ReactNode } from "react";
+import * as RadixSelect from "@radix-ui/react-select";
+import { Icon } from "../../Icon/Icon";
+import {
+  Error,
+  FormElement,
+  FormRoot,
+  ItemSeparator,
+  Label,
+  OptionContainer,
+} from "../commonElement";
+import { uniqueId } from "lodash";
+import styled from "styled-components";
+
+interface SelectProps {
+  placeholder?: ReactNode;
+  label: ReactNode;
+  children: ReactNode;
+  error?: ReactNode;
+}
+type Props = RadixSelect.SelectProps &
+  Omit<HTMLAttributes<HTMLDivElement>, "children" | "placeholder"> &
+  SelectProps;
+
+const SelectRoot = styled(RadixSelect.Root)`
+  width: 100%;
+`;
+const SelectTrigger = styled(RadixSelect.Trigger)<{ error: boolean }>`
+  width: 100%;
+  ${FormElement};
+  ${(props) =>
+    props.error
+      ? `
+      border: 1px solid var(--click-field-color-stroke-error, #C10000) !important;
+      background: var(--click-field-color-background-active, #FFF) !important;
+      & > :not(.cui-select-icon) {
+        color: var(--click-field-color-text-error, #C10000) !important;
+      }`
+      : ""}
+`;
+const SelectContent = styled(RadixSelect.Content)`
+  width: var(--radix-select-trigger-width);
+  max-height: var(--radix-select-content-available-height);
+  border-radius: 0.25rem;
+  border: 1px solid var(--click-context-menu-stroke-default, #e6e7e9);
+  background: var(--click-context-menu-color-background-default, #fff);
+  box-shadow: 0px 1px 3px 0px rgba(16, 24, 40, 0.1),
+    0px 1px 2px 0px rgba(16, 24, 40, 0.06);
+  overflow: hidden;
+  display: flex;
+  padding: 0.5rem 0rem;
+  align-items: flex-start;
+  gap: 0.625rem;
+  .cui-select-scroll-button {
+    display: flex;
+    align-items: center;
+    width: 100%;
+    justify-content: center;
+    height: 25px;
+    background-color: white;
+    color: var(--click-context-menu-color-text-default, #161517);
+    &:hover {
+      color: var(--click-context-menu-color-text-active, #161517);
+    }
+    cursor: default;
+  }
+`;
+const SelectViewport = styled(RadixSelect.Viewport)`
+  width: 100%;
+`;
+
+const Select = ({
+  placeholder = "Select an option",
+  label,
+  children,
+  disabled,
+  id,
+  error,
+  value,
+  defaultValue,
+  onValueChange,
+  open,
+  defaultOpen,
+  onOpenChange,
+  dir,
+  name,
+  required,
+  ...props
+}: Props) => {
+  id = id ?? uniqueId("select");
+  return (
+    <FormRoot {...props}>
+      {label && (
+        <Label
+          className='cui-label'
```

Comment:
I added it like this since we are deploying it to users outside ClickHouse to reuse 
If those users want to customize then we should allow this option

Reply:
i think the way to allow external customization would be with a labelStyles property instead of a global class